File: Front/ProCardapio-App/src/app/models/Restaurante.ts
import { Bebida } from "./Bebida";
import { Prato } from "./Prato";
import { RedeSocial } from "./RedeSocial";

export interface Restaurante {
  id: number;
  nome: string;
  local: string;
  garfos: number;
  imagemURL?: string;
  tipo: string;
  telefone: string;
  email: string;
  redesSociais: RedeSocial[];
  pratos: Prato[];
  bebidas: Bebida[];
}

File: Back/src/ProCardapio.Domain/Restaurante.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ProCardapio.Domain
{
    public class Restaurante
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string Nome { get; set; }

        [Required]
        [MaxLength(50)]
        public string Local { get; set; }

        public int Garfos { get; set; }

        public string ImagemURL { get; set; }

        [Required]
        public string Tipo { get; set; }

        [Required]
        public string Telefone { get; set; }

        [Required]
        public string Email { get; set; }

        public IEnumerable<RedeSocial> RedesSociais { get; set; }
        
        public IEnumerable<Prato> Pratos { get; set; }
        public IEnumerable<Bebida> Bebidas { get; set; }
    }
}